source/config/profileUpgrader.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: UTF-8 -*-
+#A part of NonVisual Desktop Access (NVDA)
+#Copyright (C) 2016 NV Access Limited
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+
+from logHandler import log
+from configSpec import latestSchemaVersion, confspec
+from configobj import flatten_errors
+from copy import deepcopy
+import profileUpgradeSteps
+
+schemaVersionKey = "schemaVersion"
+
+def upgrade(profile, validator):
+	# when profile is none or empty we can still validate. It should at least have a version set.
+	log.info("Checking schema version of config:\n%s" % profile)

```

Comment:
As discussed, there are two reasons to log the config on every load:
- Log the config before there is any chance of something going wrong while accessing the config (eg while getting the current schema version)
- Log the config so that we are able to diagnose mis-configurations of NVDA

The second of these goals is not really related to config upgrades. We will move that out of the upgrader and do it at debug level. The first goal, we will acheive by wrapping the first checks in a `try` and only logging at info level if there is an exception.